src/allmydata/client.py

```diff
@@ -658,6 +658,12 @@ def __init__(self, config, main_tub, i2p_provider, tor_provider, introducer_clie
         if webport:
             self.init_web(webport) # strports string
 
+        # TODO this may be the wrong location for now? but as temporary measure
+        # it allows us to get NURLs for testing in test_istorageserver.py. This
+        # will eventually get fixed one way or another in
+        # https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3901
+        self.storage_nurls = set()
```

Comment:
As far as where this might belong, I filed https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3931

For now, maybe a starting value of `None` is better?  I am thinking about code that looks at this value before `init_storage` gets to it and quietly concludes there are no storage NURLs - instead of somehow signaling a programming error about code that reads the value before it has actually been initialized.

Reply:
I have a follow-up branch that does stuff with this, I think it shoves it into the announcement directly.